std/datetime/date.d

```diff
@@ -9858,23 +9858,17 @@ private int cmpTimeUnitsCTFE(string lhs, string rhs) @safe pure nothrow @nogc
     import std.format : format;
     import std.meta : AliasSeq;
 
-    static string genTest(size_t index)
+    static assert(timeStrings.length == 10);
```

Comment:
This assert is not necessary. It was there in the original code only because, the author didn't use `foreach (n; staticIota!(0, timeStrings.length))` (e.g. `staticIota` from `std.typecons`).